Gulpfile.js

```diff
@@ -0,0 +1,48 @@
+const gulp = require('gulp');
+const fs = require('fs');
+const connect = require('gulp-connect');
+const jade = require('gulp-jade');
+const rename = require('gulp-rename');
+
+gulp.task('connect', () => {
+  connect.server({
+    root: 'build',
+    port: 8000,
+    livereload: true,
+  });
+});
+
+gulp.task('jade', () => {
+  gulp.src('jade/*.jade')
+    .pipe(jade({
+      locals: JSON.parse(fs.readFileSync('./data-source.json', 'utf8')),
+    }))
+    .pipe(rename({ dirname: '' }))
+    .pipe(gulp.dest('build/html'))
+    .pipe(connect.reload());
+
+  gulp.src('index.jade')
+    .pipe(jade())
+    .pipe(gulp.dest('build'));
+});
+
+gulp.task('watch-jade', () => {
+  gulp.watch(['**/*.jade'], ['jade']);
+});
+
+gulp.task('watch-data-source', () => {
+  gulp.watch('data-source.json', ['jade']);
```

Comment:
Nice. One problem with `jade --watch` is that it only responds to a Jade template (or related template) change, not to changes to json data sources. This looks like a nice solution to that!

As per above, it would be really nice to parameterize this. So when the user adds a `watchAndRenderWith(...)` entry, this `gulp.watch` action will be done on the data source.